fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -2363,9 +2379,8 @@ def get_shard_from_optim_state_dict(self, full_optim_state_dict: Dict[str, Any])
         ids_not_to_shard = copy.deepcopy(full_optim_state_dict["uncollected_local_ids"])
         if self.flatten_parameters:
             full_optim_state_dict = ou.flatten_optim_state_dict(full_optim_state_dict)
-            assert len(full_optim_state_dict["state"]) in (
-                0,
-                len(instance_list),
+            assert len(full_optim_state_dict["state"]) <= len(
```

Comment:
due to unused param, the length can be anything from 0 to the totally number of fsdp instances (because of flattening)

Reply:
Can we add a comment to say the same?

Reply:
sure